packages/db/migrations/0107.do.received_emails.sql

```diff
@@ -0,0 +1,44 @@
+-- Type: DO
+-- Name: received_emails
+-- Description: Create a table for received emails
+
+BEGIN;
+
+CREATE TABLE omnivore.received_emails (
+    id uuid PRIMARY KEY DEFAULT uuid_generate_v1mc(),
+    user_id uuid NOT NULL REFERENCES omnivore.user ON DELETE CASCADE,
+    "from" text NOT NULL,
+    "to" text NOT NULL,
+    subject text NOT NULL DEFAULT '',
+    "text" text NOT NULL,
+    html text NOT NULL DEFAULT '',
+    "type" text NOT NULL,
+    created_at timestamptz NOT NULL DEFAULT current_timestamp,
+    updated_at timestamptz NOT NULL DEFAULT current_timestamp
+);
+
+CREATE TRIGGER received_emails_modtime BEFORE UPDATE ON omnivore.received_emails
+    FOR EACH ROW EXECUTE PROCEDURE update_updated_at_column();
+
+GRANT SELECT, INSERT, UPDATE, DELETE ON omnivore.received_emails TO omnivore_user;
+
+-- Create a trigger to keep the most recent 20 emails for each user
+CREATE OR REPLACE FUNCTION omnivore.delete_old_received_emails()
```

Comment:
wild, I was wondering if this was possible

Reply:
I have tested it and this works.

Yeah, this might not be the best way if there are too many concurrent requests but since we don't need to keep exactly 20 emails per user, this might be good enough